universal-login-example/src/TokenGrantingRelayer.js

```diff
@@ -14,14 +14,30 @@ class TokenGrantingRelayer extends Relayer {
 
   addHooks() {
     this.tokenContract = new ethers.Contract(this.tokenContractAddress, Token.interface, this.deployerWallet);
-    this.hooks.addListener('created', (transaction) => {
-      waitForContractDeploy(this.provider, Token, transaction.hash)
-        .then((identityContract) => {
-          this.identityAddress = identityContract.address;
-          this.tokenContract.transfer(identityContract.address, utils.parseEther('100'));});
+    this.hooks.addListener('counterfactuallyCreated', async (identityData) => {
+      this.identityAddress = identityData.address;
+
+      const walletNonce = await this.deployerWallet.getTransactionCount();
+
+      this._giftTokens(this.identityAddress, walletNonce);
+      this._giftEth(this.identityAddress, walletNonce + 1)
+
     });
     this.hooks.addListener('added', (key) => this.tokenContract.transfer(this.identityAddress, utils.parseEther('5')));
   }
+
+  _giftTokens(identityAddress, nonce) {
+    this.tokenContract.transfer(identityAddress, utils.parseEther('100'), {
+      nonce
+    });
+  }
+
+  _giftEth(identityAddress, nonce) {
```

Comment:
We should not add ether to the newly created contract. Rather we add token, that can be later used to pay for instantiation.